packages/dynamite/dynamite/lib/src/models/type_result/some_of.dart

```diff
@@ -0,0 +1,97 @@
+part of 'type_result.dart';
+
+@immutable
+abstract class TypeResultSomeOf extends TypeResult {
+  TypeResultSomeOf(
+    super.className, {
+    required this.subTypes,
+    super.nullable,
+  }) : super(isTypeDef: false);
+
+  final Set<TypeResult> subTypes;
+
+  @override
+  String? get _builderFactory => null;
+
+  @override
+  String? get _serializer {
+    if (optimizedSubTypes.length <= 1) {
+      return null;
+    }
+
+    return '..add(${className}Extension.serializer)';
+  }
+
+  late final List<TypeResult> optimizedSubTypes = subTypes.toList();
```

Comment:
I know this is related to the follow-up PR, but please remove it from here since it is not used.